precompile/precompile.jl

```diff
@@ -4,137 +4,137 @@ function _precompile_()
     precompile(ArgParse.call, (Array{Any, 1}, Type{ArgParse.ArgParseSettings},))
     precompile(ArgParse.import_settings, (ArgParse.ArgParseSettings, ArgParse.ArgParseSettings, Bool,))
     precompile(ArgParse.override_duplicates, (Array{ArgParse.ArgParseField, 1}, ArgParse.ArgParseField,))
-    precompile(ArgParse.parse1_optarg, (ArgParse.ParserState, ArgParse.ArgParseSettings, ArgParse.ArgParseField, UTF8String, UTF8String,))
+    precompile(ArgParse.parse1_optarg, (ArgParse.ParserState, ArgParse.ArgParseSettings, ArgParse.ArgParseField, String, String,))
```

Comment:
`Compat.UTF8String` will be less likely to impact precompilation effectiveness on 0.4